pkg/kudoctl/cmd/diagnostics/runner.go

```diff
@@ -0,0 +1,32 @@
+package diagnostics
+
+// Collector - generic interface for diagnostic data collection
+// implementors are expected to return only fatal errors and handle non-fatal ones themselves
+type Collector interface {
+	Collect() error
+}
+
+// Runner - sequential runner for Collectors reducing error checking boilerplate code
+type Runner struct {
+	fatalErr error
+}
+
+func (r *Runner) Run(c Collector) *Runner {
+	if r.fatalErr == nil {
+		r.fatalErr = c.Collect()
```

Comment:
Is this correct: each `Run` call's error overrides the previous one?

Reply:
It's `until r.fatalErr != nil` keep collecting. What an I missing?

Reply:
You `Collect` one more time **after** a fatal error happened, right?

Reply:
**After** fatal error no more collectors are run... I hope.